rfcs/taskforce/0003-Integration-test-infrastructure-spec.md

```diff
@@ -0,0 +1,204 @@
+    3. Deleting and redeploying helm releases for stop and start requests from the network manager provided we can persist coda-config.
+
+##### Log engine
+
+Similar to the network manager, a log engine is also a wrapper around the logging infrastructure we'd use. The log engine will provide functions that would filter the logs from all the nodes in the network. Log engine queries correspond to the required primitives for the test DSL, for example, the `wait_for` primitive defined [here](https://github.com/CodaProtocol/coda/blob/rfc/integration-tests/docs/test_dsl_spec.md) would return when block production logs with the given filters are seen.
+
+Currently we use Stackdriver as our logging infrastructure for testnets and qa-nets. Using stackdriver, one can set up a pub/sub system to subscribe to a topic ([for more information](https://cloud.google.com/pubsub/docs/quickstart-py-mac)). Log filters can be added to a subscription and so only log messages that match the filter will be published to the topic. Each query defined in the log engine would correspond to a specific topic.
+
+Here's a rough structure of the test executive and how it links different components
+
+Assumption: (as a CI step)
+    1. generate and upload the docker image
+    2. generate genesis proof, genesis ledger with the input runtime configuration using `runtime_genesis_ledger.exe` and upload it to s3
+
+```ocaml
+
+    module type Make_test_intf = functor (Node: Node_intf) :
+        (sig
+            val execute : (module Log_engine : Log_engine_intf) -> Testnet.t -> Runtime_config.t -> bool Deferred.t
+            end)
+
+    module Test_executive
+        (Make_test : Make_test_intf)
+        ( Config : sig
+            val runtime_config : Runtime_config.t
+            val network_config: Network_config.t
+            end) =
+    struct  =
+
+        let execute () =
+            let deploy_config = {daemon_image}
+            in
+            (*deploy the network*)
+            let%bind testnet : Testnet.t = Cloud_network_manager.deploy network_config daemon_config
+            in
+            (*create the log engine based on the testnet configuration*)
+            let module Log_engine = Make_log_engine (struct
+                let testnet = testnet
+                end)
+            in
+            (*Test module for the specific Node implementation*)
+            let module Test = Make_test (Cloud_network_manager.Node)
+            in
+            (*Execute the test*)
+            let%bind result = Test.execute Log_engine testnet runtime_config
+            in
+            (*Teardown the network*)
+            let%map () = Cloud_network_manager.destroy testnet
+            in
+            if not result then failwith "Test failed" else ()
+
+    end
+```
+
+## Outstanding Questions/Concerns
+
+1. We'd still need build profiles because of the constraint constants. This means there would be a docker image uploaded for each build profile.
```

Comment:
`constraint_constants` will be moving to the general config soon, this should work automatically for tests that use `proof_level = Check`. For `proof_level = Full` tests, we will still need to handle moving the appropriate P/V keys into the docker image.